src/commercetools/testing/products.py

```diff
@@ -375,6 +379,33 @@ def updater(self, obj: dict, action: types.ProductAddPriceAction):
     return updater
 
 
+def _change_master_variant():
+    def updater(self, obj: Dict, action: types.ProductChangeMasterVariantAction):
+        new = copy.deepcopy(obj)
+        staged = action.staged
+        if staged is None:
+            staged = True
+        target_obj = _get_target_obj(new, staged)
+        master_variant = target_obj["masterVariant"]
+        if master_variant and master_variant["sku"] == action.sku:
+            return obj
+        found_sku = False
+        for variant in get_product_variants(target_obj):
+            if variant["sku"] == action.sku:
+                target_obj["variants"].append(target_obj["masterVariant"])
+                target_obj["masterVariant"] = variant
+                target_obj["variants"].remove(variant)
```

Comment:
Isn't the master variant also part of the variants list?

Reply:
No, which is why we have this kind of code everywhere;

```
def all_variants(data: types.ProductData) -> List[types.ProductVariant]:
    variants = [data.master_variant]
    if data.variants:
        variants += data.variants
    return variants
```